keymanager/src/policy/cached.rs

```diff
@@ -55,40 +55,56 @@ impl Policy {
     ///
     /// The policy is presumed trustworthy, so it's up to the caller to verify it against
     /// the consensus layer state. Empty polices are allowed only in unsafe builds.
-    pub fn init(&self, policy: Option<SignedPolicySGX>) -> Result<Vec<u8>> {
+    pub fn init(
+        &self,
+        policy: Option<SignedPolicySGX>,
+        next_policy: Option<SignedPolicySGX>,
+    ) -> Result<Vec<u8>> {
         // If this is an insecure build, don't bother trying to apply any policy.
         if policy.is_none() && Self::unsafe_skip() {
             return Ok(vec![]);
         }
 
         // Cache the new policy.
         let policy = policy.ok_or(KeyManagerError::PolicyRequired)?;
-        let new_policy = CachedPolicy::parse(policy)?;
+        let policy = CachedPolicy::parse(policy)?;
 
         // Lock as late as possible.
         let mut inner = self.inner.write().unwrap();
 
         // Compare the new serial number with the old serial number, ensure
         // it is greater.
         if let Some(old_policy) = inner.policy.as_ref() {
-            match old_policy.serial.cmp(&new_policy.serial) {
+            match old_policy.serial.cmp(&policy.serial) {
                 Ordering::Greater => return Err(KeyManagerError::PolicyRollback.into()),
                 Ordering::Equal => {
-                    if old_policy.checksum != new_policy.checksum {
+                    if old_policy.checksum != policy.checksum {
                         // Policy should be identical.
                         return Err(KeyManagerError::PolicyChanged.into());
                     }
-                    return Ok(new_policy.checksum);
+                    return Ok(policy.checksum);
```

Comment:
Should this check if there is a next policy? Otherwise the below will not execute until the next epoch.

Reply:
Nice catch, I removed this line so now the policy is in this case replaced with identical one.